pkg/controller/util.go

```diff
@@ -1188,11 +1188,17 @@ func handleFailedPod(err error, podName string, pvc *v1.PersistentVolumeClaim, r
 	// Error handling to fine-tune the event with pertinent info
 	if errQuotaExceeded(err) {
 		reason = ErrExceededQuota
-	} // TODO: Add more error cases
+	}
 
 	recorder.Event(pvc, v1.EventTypeWarning, reason, msg)
 
+	isCloneSourcePod := createCloneSourcePodName(pvc) == podName
+	if isCloneSourcePod {
+		AddAnnotation(pvc, AnnCloneSourcePodFailed, "true")
+	}
 	AddAnnotation(pvc, AnnRunningCondition, "false")
+	AddAnnotation(pvc, AnnRunningConditionReason, "")
```

Comment:
We should tell the user why we are not running. We have several options for the reason above set in `reason` we should probably use it.